server/embedded/src/org/labkey/embedded/EmbeddedExtractor.java

```diff
@@ -262,11 +266,20 @@ public void extractExecutableJar(File destDirectory, boolean remotePipeline)
                             {
                                 extractFile(in, new File(pipelineLib, "servletApi.jar"));
                             }
+                            missingServletApiJar = false;
                         }
                     }
                 }
 
-                if (!foundDistributionZip)
+                if (missingDistributionZip)
+                {
+                    throw new ConfigException("Unable to find distribution zip required to run LabKey Server.");
+                }
+                if (missingBootstrapJar)
+                {
+                    throw new ConfigException("Unable to find labkeyServer.jar required to run LabKey Server.");
+                }
+                if (missingServletApiJar)
                 {
                     throw new ConfigException("Unable to find distribution zip required to run LabKey Server.");
                 }
```

Comment:
This is the wrong error message
```suggestion
                if (missingServletApiJar)
                {
                    throw new ConfigException("Unable to find servletApi.jar required to run LabKey Remote Pipeline.");
                }
```

Reply:
Updated via a separate commit